### PR TITLE
add deposit on consensus

### PR DESCRIPTION
**Pull request by EduardoRFS**
## Depends

- [x] #78 

## Problem

As per #34 we need to be able to move tickets to the sidechain.

## Solution

Add a deposit entrypoint on the contract, this should be detected by the node and calls `Ledger.deposit` when the operation is added to a block.

## Related

- #34

### CODE REVIEW

File: tezos_interop/consensus.mligo [hunk 1]
```diff
@@ -116,15 +116,45 @@ let root_hash_main
...
 
+(* vault contract *)
+type vault_storage = (address * bytes, bytes ticket) big_map
+type vault_deposit = {
+  ticket: bytes ticket;
+  (* WARNING: deposit address should be a valid sidechain address *)
+  address: address
+}
+let vault_deposit (deposit: vault_deposit) (vault: vault_storage) =
+  let (content, ticket) =  Tezos.read_ticket deposit.ticket in
```
Comment (anchpop) [inline tezos_interop/consensus.mligo:127]:
```suggestion
  let (content, ticket) = Tezos.read_ticket deposit.ticket in
```

**Review by anchpop — APPROVED** [review]

File: tezos_interop/consensus.mligo (hunk 1 above)
Comment (EduardoRFS) [reply tezos_interop/consensus.mligo]: thanks

**Review by callistonianembrace — APPROVED** [review]
I have a very shallow understanding of these contracts. I need to study them better before I can provide meaningful contract. Approving to unblock merging.

**Comment by anchpop** [comment]
@callistonianembrace Me too, let me know if you want to go over them together (or if you go over them with @EduardoRFS, let me know so I can join in)

**Pull request merged**